src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -202,25 +231,100 @@ private Tooltip createTitleTooltip(TurboIssue issue) {
         return titleTooltip;
     }
 
-    private FlowPane createTopPane() {
-        FlowPane topPane = new FlowPane();
-        topPane.setPadding(new Insets(20, 0, 10, 0));
-        topPane.setHgap(5);
-        topPane.setVgap(5);
-        return topPane;
+    private void setTitleLabel(TurboIssue issue) {
+        title.setText((issue.isPullRequest() ? "PR #" : "Issue #") 
+            + issue.getId() + ": " + issue.getTitle());
+    }
+
+    private Label createTextLabel(String input) {
+        Label label = new Label(input);
+        label.setPadding(new Insets(2, 5, 2, 5));
+        return label;
+    }
+
+
+    private Label createGroupTitle(String name) {
+        Label groupName = new Label(name);
+        groupName.setPadding(new Insets(0, 5, 5, 0));                                     
+        groupName.setMaxWidth(ELEMENT_MAX_WIDTH - 10);                                    
+        groupName.setStyle("-fx-font-size: 110%; -fx-font-weight: bold;");            
+        return groupName;
     }
 
-    private TextField createTextField() {
-        TextField textField = new TextField();
-        textField.setId("labelPickerTextField");
-        textField.setPrefColumnCount(30);
-        return textField;
+    private FlowPane createGroupPane(Insets padding) {
+        FlowPane group = new FlowPane();                                                
+        group.setHgap(5);
+        group.setVgap(5);
+        group.setPadding(padding);
+        return group;
     }
 
-    private VBox createBottomBox() {
-        VBox bottomBox = new VBox();
-        bottomBox.setPadding(new Insets(10, 0, 0, 0));
-        return bottomBox;
+    // Event handling 
+
+    private void setupEvents(Stage stage) {
+        showingProperty().addListener(e -> {
+            positionDialog(stage);
+        });
+    }
+
+    private void registerQueryHandler() {
+        queryField.textProperty().addListener((observable, oldValue, newValue) -> {
+            LabelPickerState state = uiLogic.determineState(
+                new LabelPickerState(new HashSet<String>(issue.getLabels())), 
+                repoLabelsSet, queryField.getText().toLowerCase());
+            finalAssignedLabels = state.getAssignedLabels();
+            populatePanes(state.getMatchedLabels(), state.getCurrentSuggestion());
+        });
+    }
+
+    private void positionDialog(Stage stage) {
+        if (getDialogHeight().isPresent()) {
+            setX(stage.getX() + stage.getScene().getX());
+            setY(stage.getY() +
+                 stage.getScene().getY() +
+                 (stage.getScene().getHeight() - getHeight()) / 2);
+        } 
+    }
+
+    // TODO refactor if else 
```

Comment:
Sorry when you say removed and dealt with, do you mean the updateAssignedLabels not the positionDialog right ? (referring to the TODO comment)